### PR TITLE
How to build an Uberjar from containers

### PR DESCRIPTION
**Pull request by dragonsahead**
This is a simple trick that I got from @dacort which creates a container with scratch with the uberjar, so now just doing
```
DOCKER_BUILDKIT=1 docker build -f Dockerfile_export_jar --output target/uberjar/ .
```
will get you an uberjar with a clean, reproducible and fully contained environment (the builder stage)

This can be a great solution for anyone that wants to build Metabase without installing anything locally or even a one liner for our release process

### CODE REVIEW

**Review by camsaul — APPROVED** [review]

**Comment by paoliniluis** [comment]
@ariya I added some notes to the documentation, I know that your concern might be that we have another Dockerfile to support, but I'm sure that if we change the release process to streamline it with containers we will only have 2: this one that will output a JAR) and maybe another one that will put that JAR file inside a runner (might be a simple shell script that compiles and exports a JAR and another one that bundles the JAR into many runners depending on the architecture, a simple variation of the current one). If you believe so, I can merge this

**Comment by paoliniluis** [comment]
Did a very tiny change in the doc, that would be enough for exporting the entire container into the host filesystem so you can get the JAR from /app/

File: docs/developers-guide/build.md [hunk 1]
```diff
@@ -164,3 +164,11 @@ The entire Metabase application is compiled and assembled into a single .jar fil
     ./bin/build
 
 After running the build script simply look in `target/uberjar` for the output .jar file and you are ready to go.
+
+### Build the Uberjar on a containerized environment
```
Comment (jeff-bruemmer) [inline docs/developers-guide/build.md:168]:
```suggestion
### Build  a Metabase Uberjar in a containerized environment
```

File: docs/developers-guide/build.md [hunk 2]
```diff
@@ -164,3 +164,11 @@ The entire Metabase application is compiled and assembled into a single .jar fil
...
+
+If you want to build Metabase but you don't want to install Clojure, Java and Node.js on your host machine, you can build the Uberjar inside a container by just simply running:
```
Comment (jeff-bruemmer) [inline docs/developers-guide/build.md:170]:
```suggestion
If you want to build Metabase without installing Clojure, Java, and Node.js on your host machine, you can build the Uberjar inside a container by running:
```

File: docs/developers-guide/build.md [hunk 3]
```diff
@@ -164,3 +164,11 @@ The entire Metabase application is compiled and assembled into a single .jar fil
...
+```
+DOCKER_BUILDKIT=1 docker build --output container-output/ .
+```
+and you will have a metabase.jar on the app/ folder ready to run.
```
Comment (jeff-bruemmer) [inline docs/developers-guide/build.md:174]:
```suggestion
Make sure that your Docker Daemon is running before executing the command. After running the command, you'll find the Metabase JAR file at `./container-output/app/metabase.jar`.
```

**Review by jeff-bruemmer — APPROVED** [review]

**Pull request merged**